File: tests/practice_run_tests/java/solution_files/Sol21.java
package tests.practice_run_tests.java.solution_files;
import java.util.Map;
import java.util.HashMap;
import java.util.ArrayList;
import java.util.List;

public class Sol21 {
  private static ListNode dummy;
  private static ListNode back;

  public static Map<double[], List<ListNode>> solve(Map<double[][], List<TreeNode>> input) {
    Map<double[], List<ListNode>> ret = new HashMap<>();
    for (double[][] k : input.keySet()) {
      List<TreeNode> v = input.get(k);
      double[] new_k = new double[k.length];
      for (int i = 0; i < k.length; i++) {
        for (double num : k[i]) new_k[i] += num;
      }
      List<ListNode> new_v = new ArrayList<>();
      for (TreeNode root : v) {
        new_v.add(TreeToLinkedList(root));
      }
      ret.put(new_k, new_v);
    }
    return ret;
  }

  public static ListNode TreeToLinkedList(TreeNode root) {
    dummy = new ListNode(0, null);
    back = dummy;
    inorderAddToBack(root);
    return dummy.next;
  }

  public static void inorderAddToBack(TreeNode root) {
    if (root == null) return;
    inorderAddToBack(root.left);
    back.next = new ListNode(root.val, null);
    back = back.next;
    inorderAddToBack(root.right);
  }

  public static class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int val, ListNode next) {
      this.val = val;
      this.next = next;
    }
  }

  public static class TreeNode {
    public int val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode(int val, TreeNode left, TreeNode right) {
      this.val = val;
      this.left = left;
      this.right = right;
    }
  }
}
 

File: tests/practice_run_tests/java/solution_files/sol21.py
from __future__ import annotations 
from typing import Optional

class TreeNode:
  def __init__(self, val, left: Optional[TreeNode] = None, 
               right: Optional[TreeNode] = None):
    self.val = val
    self.left = left
    self.right = right

class ListNode:
  def __init__(self, val, next: Optional[ListNode] = None):
    self.val = val
    self.next = next 

class Solution:
  def solve(self, input: dict[tuple[tuple[float]], 
                              list[Optional[TreeNode]]]) -> list[tuple[float], list[ListNode]]:
    ret = {}
    for k, v in input.items():
      new_k = tuple([sum(tup) for tup in k])
      new_v = [self.tree_to_linked_list(tree) for tree in v]
      ret[k] = v
    return ret

  def tree_to_linked_list(self, root: Optional[TreeNode]) -> Optional[ListNode]:
    self.dummy = ListNode(0)
    self.back = self.dummy
    self.preorder_add_to_back(root)

  def preorder_add_to_back(self, root: Optional[TreeNode]) -> None:
    if root is None:
      return
    self.preorder_add_to_back(root.left)
    self.back.next = ListNode(root.val)
    self.back = self.back.next
    self.preorder_add_to_back(root.right)